modules/idTestingModule/index.js

```diff
@@ -0,0 +1,119 @@
+  moduleConfig = config.getConfig(MODULE_NAME) || {};
+  const {suppression, testRun, modules = [], storeSplits} = moduleConfig;
+  let testConfig;
+
+  if (testRun && storeSplits && storeSplits !== storeSplitsMethod.PAGE) {
+    testConfig = getStoredTestConfig(testRun, storeSplits, storageManager);
+    if (!testConfig || !compareConfigs(testConfig, modules)) {
+      testConfig = internals.getCalculatedSubmodules();
+      storeTestConfig(testRun, testConfig, storeSplits, storageManager);
+    }
+  }
+
+  testConfig = testConfig ?? internals.getCalculatedSubmodules();
+
+  const bannedModules = testConfig.filter(({enabled}) => !enabled);
+
+  if (bannedModules.length) {
+    const init = suppression === suppressionMethod.SUBMODULES;
+    rules.push(registerActivityControl(ACTIVITY_ENRICH_EIDS, MODULE_NAME, userIdSystemBlockRule(bannedModules, init)));
+  }
+
+  if (testRun) {
+    setAnalyticLabels({[testRun]: testConfig});
+  }
+}
+
+export function reset() {
+  rules.forEach(unregister => unregister());
+  setAnalyticLabels({});
+  rules = [];
+}
+
+function compareConfigs(old, current) {
+  return deepEqual(old.map(({percentage}) => percentage), current.map(({percentage}) => percentage));
+}
+
+function userIdSystemBlockRule(bannedModules, init) {
+  return (params) => {
+    if (params.init === init && params[ACTIVITY_PARAM_COMPONENT_TYPE] === MODULE_TYPE_UID && bannedModules.find(({name}) => name === params[ACTIVITY_PARAM_COMPONENT_NAME])) {
+      return {allow: false, reason: 'disabled due to AB testing'};
+    }
+  }
+};
+
+export function getCalculatedSubmodules() {
+  const {modules = []} = moduleConfig;
+  return modules
+    .map(({name, percentage}) => {
+      const enabled = Math.random() < percentage;
+      return {name, percentage, enabled}
+    });
+};
+
+export function getStoredTestConfig(testRunString, storeSplits, storageManager) {
+  const [checkMethod, getMethod] = {
+    [storeSplitsMethod.SESSION]: [storageManager.sessionStorageIsEnabled, storageManager.getDataFromSessionStorage],
+    [storeSplitsMethod.LIFE_OF_USER]: [storageManager.localStorageIsEnabled, storageManager.getDataFromLocalStorage],
+  }[storeSplits];
+
+  if (!checkMethod()) {
+    logError(`${MODULE_NAME} Unable to save testing module config - storage is not enabled`);
+  }
+
+  try {
+    return JSON.parse(getMethod(testRunString));
+  } catch {
+    return null;
+  }
+};
+
+export function storeTestConfig(testRunString, modules, storeSplits, storageManager) {
+  const [checkMethod, storeMethod] = {
+    [storeSplitsMethod.SESSION]: [storageManager.sessionStorageIsEnabled, storageManager.setDataInSessionStorage],
+    [storeSplitsMethod.LIFE_OF_USER]: [storageManager.localStorageIsEnabled, storageManager.setDataInLocalStorage],
+  }[storeSplits];
+
+  if (!checkMethod()) {
+    logError(`${MODULE_NAME} Unable to save testing module config - storage is not enabled`);
+  }
+
+  storeMethod(testRunString, JSON.stringify(modules));
```

Comment:
if `testRunString` changes (meaning we are now running a new test) we should get rid of the old stored config.